spec.bs

```diff
@@ -3808,32 +3864,65 @@ The <dfn for=Navigator method>getInterestGroupAdAuctionData(|configIDL|)</dfn> m
         1. Set |requestSize| to |requestSize| + |buyerConfig|'s [=auction data buyer config/size=].
-    1. Set |config|'s [=auction data config/request size=] to |requestSize|.
-  1. Return |config|.
+  1. If |configIDL|["{{AdAuctionDataConfig/seller}}"] [=map/exists=]:
+    1. Let |seller| be the result of running [=parse an https origin=] on
+       |configIDL|["{{AdAuctionDataConfig/seller}}"].
+    1. If |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"] [=map/exists=]:
+      1. Let |coordinator| be the result of running [=parse an https origin=] on
+        |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"].
+    1. Otherwise
+      1. Let |coordinator| be the [=user agent=]'s default Bidding and Auction Services coordinator.
+    1. If |seller| or |coordinator| are failure, then [=exception/throw=] a {{TypeError}}.
+    1. Let |config| be a new [=auction data config=] with the following [=struct/items=]:
+      : [=auction data config/publisher=]
+      :: |top_level_origin|
+      : [=auction data config/seller=]
+      :: |seller|
+      : [=auction data config/coordinator=]
+      :: |coordinator|
+      : [=auction data config/request size=]
+      :: |requestSize|
+      : [=auction data config/per buyer config=]
+      :: |perBuyerConfigs|
+    1. [=list/Append=] |config| to |configs|.
+  1. Otherwise:
+    1. If |configIDL|["{{AdAuctionDataConfig/sellers}}"] does not [=map/exist=], then [=exception/throw=] a {{TypeError}}.
+    1. If |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"] [=map/exists=], then [=exception/throw=] a {{TypeError}}.
+    1. [=list/For each=] |sellerConfig| in |configIDL|["{{AdAuctionDataConfig/sellers}}"]:
+      1. Let |seller| be the result of running [=parse an https origin=] on
+        |sellerConfig|["{{AdAuctionOneSeller/seller}}"].
+      1. If |configs| [=list/contains=] an [=auction data config=] whose
+         [=auction data config/seller=] is equal to |seller|, then
+         [=exception/throw=] a {{TypeError}}.
+      1. If |sellerConfig|["{{AdAuctionOneSeller/coordinatorOrigin}}"] [=map/exists=]:
+        1. Let |coordinator| be the result of running [=parse an https origin=] on
+          |sellerConfig|["{{AdAuctionOneSeller/coordinatorOrigin}}"].
+      1. Otherwise:
+        1. Let |coordinator| be the [=user agent=]'s default Bidding and Auction Services coordinator.
+      1. If |seller| or |coordinator| are failure, then [=exception/throw=] a {{TypeError}}.
+      1. Let |config| be a new [=auction data config=] with the following [=struct/items=]:
+        : [=auction data config/publisher=]
+        :: |top_level_origin|
+        : [=auction data config/seller=]
+        :: |seller|
+        : [=auction data config/coordinator=]
+        :: |coordinator|
+        : [=auction data config/request size=]
+        :: |requestSize|
+        : [=auction data config/per buyer config=]
+        :: |perBuyerConfigs|
+      1. [=list/Append=] |config| to |configs|.

```

Comment:
instead of this, probably slightly better to change [=auction data config=]'s seller and coordinator to a list of (seller, coordinator) (probably a new struct), so that it's clearer that all other fields are shared for all sellers. Won't let it block this PR though.

Reply:
I can't do that without also changing the IETF spec, since the structure is used by that spec.

Reply:
sure. That can be changed later if we want to, but no strong opinion that we must do.